clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -728,6 +730,27 @@ public static void setDebugLevel(LogLevel level) {
         debugLevel = level.intValue();
     }
 
+    /**
+     * Use this method to set a custom locale for the CleverTapAPI
+     *
+     * @param locale - The custom locale to be set
+     */
+    @SuppressWarnings({"unused"})
+    public static void setLocale(String locale) {
+        if(TextUtils.isEmpty(locale))
+            Logger.i("Empty Locale provided for setCustomLocale");
+        customLocale = locale;
```

Comment:
@Anush-Shand instead of setting `customLocale` as a variable in this class, you can expose a setter inside `DeviceInfo` class
This way you can get the Locale from `DeviceInfo` inside the `CTJsonConverter` class